qiskit/transpiler/passes/layout/sabre_pre_layout.py

```diff
@@ -0,0 +1,213 @@
+
+    """
+
+    def __init__(
+        self,
+        target=None,
+        coupling_map=None,
+        max_distance=2,
+        error_rate=0.1,
+        max_trials_vf2=100,
+        improve_layout=True,
+    ):
+        """SabrePreLayout initializer.
+
+        The pass works by augmenting the coupling map with more and more "extra" edges
+        until VF2 succeeds to find a perfect graph isomorphism. More precisely, the
+        augmented coupling map contains edges between nodes that are within a given
+        distance ``d`` in the original coupling map, and the value of ``d`` is increased
+        until an isomorphism is found.
+
+        Intuitively, a better layout involves fewer extra edges. The pass also optionally
+        minimizes the number of extra edges involved in the layout until a local minimum
+        is found. This involves removing extra edges and running VF2 to see if an
+        isomorphism still exists.
+
+        Args:
+            target (Target): A target representing the backend device. If specified, it will
+                supersede a set value for ``coupling_map``.
+            coupling_map (CouplingMap): directed graph representing the original coupling map.
+            max_distance (int): the maximum distance to consider for augmented coupling maps.
+            error_rate (float): the error rate to assign to the "extra" edges. A non-zero
+                error rate prioritizes VF2 to choose original edges over extra edges.
+            max_trials_vf2 (int): specifies the maximum number of VF2 trials. A larger number
+                allows VF2 to explore more layouts, eventually choosing the one with the smallest
+                error rate.
+            improve_layout (bool): whether to improve the layout by minimizing the number of
+                extra edges involved. This might be time-consuming as this requires additional
+                VF2 calls.
+
+        Raises:
+            TranspilerError: At runtime, if neither ``coupling_map`` or ``target`` are provided.
+        """
+
+        self.target = target
+        self.coupling_map = coupling_map
+        self.max_distance = max_distance
+        self.error_rate = error_rate
+        self.max_trials_vf2 = max_trials_vf2
+        self.improve_layout = improve_layout
+
+        if self.target is not None:
+            self.coupling_map = self.target.build_coupling_map()
+
+        super().__init__()
+
+    def run(self, dag):
+        """Run the SabrePreLayout pass on `dag`.
+
+        The discovered starting layout is written to the property set
+        value ``sabre_starting_layouts``.
+
+        Args:
+            dag (DAGCircuit): DAG to create starting layout for.
+        """
+
+        if self.coupling_map is None:
+            raise TranspilerError(
+                "SabrePreLayout requires either target or coupling_map to be provided."
+            )
+
+        starting_layout = None
+        cur_distance = 1
+        while True:
```

Comment:
Should this just be?:

```suggestion
        while cur_distance <= self.max_distance:
```

and then drop the if condition at the end of the loop.

When I first saw this I was just worried about a potential infinite loop for something running VF2 :)

Reply:
Oh, of course :). Done in [f006dcd](https://github.com/Qiskit/qiskit/pull/10829/commits/f006dcd67978b7d0e1719428ec90aaf54053015f). Leftover from when the default value was `None`.